Fix work packages menu item never being selected

mfrister opened the pull request:
https://www.openproject.org/work_packages/15441

For query menu items this already worked, but the work packages main
item was never selected.

mfrister committed:
Fix work packages menu item never being selected

For query menu items this already worked, but the work packages main
item was never selected.

TeatroIO commented:
I've prepared a stage. <a href='http://fix-wp-menu-item-not-selected.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

tessi commented on app/assets/javascripts/angular/layout/query-menu-items.js:
```diff
@@ -52,7 +52,12 @@ angular.module('openproject.layout')
+        // Apparently the queryId sometimes is an number, sometimes a string, sometimes
+        // undefined and sometimes null. String ensures number and string comparisons work,
+        // '|| null' ensures null and undefined are regarded as the same.
+        // No idea though, why these sometimes are null and sometimes are undefined.
```
I suggest to ask @myabc on that matter. He probably has some insight on this mysterious case.

myabc replied:
To be honest I'm not sure why it's sometimes `null` and sometimes `undefined`.

Rather than having to mess around with explicit coercion, I guess I'd probably use the double equals operator here: `scope.queryId == $stateParams.query_id`

tessi commented on karma/tests/layout/query-menu-item-factory-test.js:
```diff
@@ -113,4 +113,65 @@ describe('queryMenuItemFactory', function() {
+    var path = '/work_packages',
+        title = 'Work Packages',
+        objectId = undefined,
+        queryId = 1;
```
You don't use that variable :)

mfrister replied:
Thanks, fixed :)

tessi commented:
Now it selects the WorkPackage menu item even when you visit other areas of a Project. This is an example when looking at the project overview:

![selection_057](https://cloud.githubusercontent.com/assets/206108/4573163/42425336-4f8f-11e4-9f20-a1b2f8a9a849.png)

mfrister committed:
Fix work packages menu item always being selected

This adds a check to only mark query menu items when being on the work
packages page.

Not very pretty, but I couldn't think of a better way to distinguish
work package pages from others.

mfrister committed:
Query Menu Items JS: Simplify comparison

mfrister commented:
Oops. Fixed that and simplified the comparison as @myabc suggested.

myabc commented:
:+1:
